components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesStorage.kt

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.top.sites
+
+/**
+ * Abstraction layer above the [PinnedSiteStorage] and [PlacesHistoryStorage] storages.
+ */
+interface TopSitesStorage {
+    /**
+     * Return a unified list of top sites based on the given number of sites desired.
+     * If `includeFrecent` is true, fill in any missing top sites with frecent top site results.
+     */
+    suspend fun getTopSites(totalNumberOfSites: Int, includeFrecent: Boolean): List<TopSite>
```

Comment:
👍 

Just adding a not that this would resolve: https://github.com/mozilla-mobile/android-components/issues/7459